Replace how to upload images with swiftui views

bjtitus opened the pull request:
Contribution from @teaseaque in #2967 

Fixes #7 

- Added 2 SwiftUI views that support themes and text localization to replace 2 images including hardcoded English text on HowToUploadViewController sheet
- Removed hardcoded English image assets and references since they are no longer used to reduce bundle size

## Before testing
If the test device system language is English, add a non-English language that is supported by Pocket Casts (e.g. French) in Settings app > General > Language & Region. Then, set the preferred language for Pocket Casts to the non-English language in Settings app > Apps > Pocket Casts

## Test
1. Run this branch
2. Navigate to HowToUploadViewController sheet via Profile > Files > How do I do that?
3. Observe the 2 images between instructions are in the preferred language
4. Change the Pocket Casts theme and repeat steps 2 & 3

## Checklist

- [X] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [X] I have considered adding unit tests for my changes.
- [X] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

teaseaque committed:
Added SwiftUI views and localization for text subviews to improve acessibility for images used in HowToUploadViewController

teaseaque committed:
Removed HowToUploadViewController images and references replaced with SwiftUI views

teaseaque committed:
Added new view files to pbxproj

teaseaque committed:
Adjust hover circle x offset

Increase hover circle x offset 1 point to the right

teaseaque committed:
Clarified image subview names

Clarified image subview names

teaseaque committed:
Add didMove for image2HostVC

Co-authored-by: Brandon Titus <b@titus.io>

teaseaque committed:
Add didMove for image1HostVC

Co-authored-by: Brandon Titus <b@titus.io>

teaseaque committed:
Simplified HowToUploadViewController using SwiftUI and refactored images into ViewBuilders

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is larger than 500 lin...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

bjtitus committed:
Merge branch 'trunk' into teaseaque/replaced-how-to-upload-images-with-swiftui-views

# Conflicts:
#	podcasts/Strings+Generated.swift
#	podcasts/UploadedViewController.swift

bjtitus committed:
Fix generated strings from merge

bjtitus committed:
Fix linter error

bjtitus committed:
Update HotToUpload/Share names for better clarity

Instead of 1 and 2, let's refer to these by their contents for share menu and action instead.

bjtitus committed:
Add changelog entry for How to Upload localization

bjtitus commented:
This has already been reviewed in https://github.com/Automattic/pocket-casts-ios/pull/2967 but can you give it a once over @danielebogo?

danielebogo reviewed (commented):
Hey @bjtitus for these 2 key that have been changed do we need to wait for Glot press to be re-translated?: `how_to_upload_second_instruction` and `how_to_upload_first_instruction`

Also we can run CI on the external contributor PR directly

<img width="1179" height="2556" alt="simulator_screenshot_890A9C14-2EA6-4E1E-88B1-AA7AC6B65F14" src="https://github.com/user-attachments/assets/e4045a3a-e8e7-49a4-ae78-75231a6e3689" />
